assets/app/scripts/controllers/pods.js

```diff
@@ -11,39 +11,19 @@ angular.module('openshiftConsole')
   .controller('PodsController', function ($scope, $http, DataService) {
     $scope.pods = {};
     $scope.podsByLabel = {};
-
-    var callback = function(action, pod) {
+    $scope.watches = [];
+    var callback = function(pods, action, pod) {
       $scope.$apply(function() {
-        var label, labelValue;
-        if (action === 'ADDED' || action === 'MODIFIED') {
-          $scope.pods[pod.id] = pod;
-
-          for (label in pod.labels) {
-            labelValue = pod.labels[label];
-            if (!$scope.podsByLabel[label]) {
-              $scope.podsByLabel[label] = {};
-            }
-            if (!$scope.podsByLabel[label][labelValue]) {
-              $scope.podsByLabel[label][labelValue] = {};
-            }
-            $scope.podsByLabel[label][labelValue][pod.id] = pod;
-          }
-
-        }
-        else if (action === 'DELETED') {
-          delete $scope.pods[pod.id];
-
-          for (label in pod.labels) {
-            labelValue = pod.labels[label];
-            if ($scope.podsByLabel[label] && $scope.podsByLabel[label][labelValue]) {
-              delete $scope.podsByLabel[label][labelValue][pod.id];
-            }
-          }        
-        }
+        $scope.pods = pods.by("id");
+        $scope.podsByLabel = pods.by("labels", "id");
       });
     };
 
-    // TODO we should separately call DataService.getList first to make a static request
-    // But this won't help us until we implement tracking of resourceVersion
-    DataService.subscribe("pods", callback, $scope);
+    $scope.watches.push(DataService.watch("pods", $scope, callback));
+
+    $scope.$on('$destroy', function(){
+      for (var i = 0; i < $scope.watches.length; i++) {
+        DataService.unwatch($scope.watches[i]);
+      }
+    });    
```

Comment:
Do we need to clear $scope.watches when we're done to avoid memory leaks or accidental future reuse?